app/tests/spec/lib/logger.js

```diff
@@ -0,0 +1,132 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define(function (require, exports, module) {
+  'use strict';
+
+  var AuthErrors = require('lib/auth-errors');
+  var chai = require('chai');
+  var Logger = require('lib/logger');
+  var sinon = require('sinon');
+  var WindowMock = require('../../mocks/window');
+
+  var assert = chai.assert;
+
+  describe('lib/logger', function () {
+    var logger;
+
+    describe('constructor', function () {
+      it('should create logger', function () {
+        sinon.spy(window.console, 'info');
+
+        var information = 'Hello Firefox!';
+
+        logger = new Logger();
+        logger.info(information);
+
+        assert.equal(window.console.info.args[0][0], information);
+        window.console.info.restore();
+      });
+
+      it('should create logger with window', function () {
+        var mockWindow = new WindowMock();
+        var information = 'Hello Firefox!';
+        sinon.spy(mockWindow.console, 'info');
+
+        logger = new Logger(mockWindow);
+        logger.info(information);
+
+        assert.equal(mockWindow.console.info.args[0][0], information);
+        mockWindow.console.info.restore();
+      });
+    });
+
+    describe('info', function () {
+      beforeEach(function () {
+        logger = new Logger(window);
+        sinon.spy(window.console, 'info');
+      });
+
+      afterEach(function () {
+        window.console.info.restore();
+      });
+
+      it('should log from string', function () {
+        var information = 'Information about stuff!';
+        logger.info(information);
+
+        assert.equal(window.console.info.args[0][0], information);
```

Comment:
Should we also assert `window.console.info.thisValues[0]`? If applying the invocations to the console object is required for them to work correctly, it feels like we should.